### PR TITLE
Flattening sets out of data_exporter output

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

* This PR flattens the output of the data_exporter so that items in a set can be seen and their credit metadata can be looked up. Hurrah! (Yes, it would be better to pull data directly from the workspace instead of going through the setapi nonsense... a PR for that will be coming soon.)
* Added some tests for data_exporter.py since there were none.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-X
- [ ] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (Python) I have run Ruff on changed Python code
- [ ] If appropriate, I have recompiled the app and added the updated `StaticNarrativeImpl.py`, `StaticNarrativeServer.py`, and `compile_report.json` to this PR.

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: lib/StaticNarrative/exporter/data_exporter.py [hunk 1]
```diff
@@ -98,12 +89,28 @@ def export_narrative_data(
...
-def _reshape_obj(obj_info: list[str | dict[str, Any]], obj_upa: str) -> list[str | dict[str, Any]]:
+def _index_obj(
+    item: dict[str, Any], type_info: dict[str, Any], indexed_data: dict[str, Any]
+) -> None:
+    """Index by object ID and save the type information."""
+    obj = item[OBJ_INFO]
+    obj_type = obj[2].split("-")[0]
+    if obj_type in IGNORED_TYPES:
+        return
+    obj_upa = generate_upa(item[OBJ_INFO])
```
Comment (briehl) [inline lib/StaticNarrative/exporter/data_exporter.py]: You already have `obj` defined as that above, might as well use it!

File: lib/StaticNarrative/exporter/data_exporter.py [hunk 2]
```diff
@@ -98,12 +89,28 @@ def export_narrative_data(
     return output_data
 
 
-def _reshape_obj(obj_info: list[str | dict[str, Any]], obj_upa: str) -> list[str | dict[str, Any]]:
+def _index_obj(
+    item: dict[str, Any], type_info: dict[str, Any], indexed_data: dict[str, Any]
+) -> None:
+    """Index by object ID and save the type information."""
```
Comment (briehl) [inline lib/StaticNarrative/exporter/data_exporter.py]: May want to note for posterity that this has the side effects of updating the input dicts.

**Review by briehl — CHANGES_REQUESTED** [review]
A couple minor points in the `_index_obj` function.

File: lib/StaticNarrative/exporter/data_exporter.py (hunk 1 above)
Comment (ialarmedalien) [reply lib/StaticNarrative/exporter/data_exporter.py]: Doh!

**Review by briehl — APPROVED** [review]
LGTM

**Pull request merged**